include/zephyr/posix/posix_types.h

```diff
@@ -109,6 +109,12 @@ typedef struct pthread_once pthread_once_t;
 /* Newlib typedefs pthread_once_t as a struct with two ints */
 BUILD_ASSERT(sizeof(pthread_once_t) >= sizeof(struct pthread_once));
 
+/* PID */
+#if defined(CONFIG_MINIMAL_LIBC) || defined(CONFIG_ARMCLANG_STD_LIBC) \
+	|| defined(CONFIG_ARCMWDT_LIBC)
+typedef int32_t pid_t;
+#endif
+
```

Comment:
Probably needs a rebase and I think this can be removed

Reply:
done